app/models/spree/image_decorator.rb

```diff
@@ -0,0 +1,22 @@
+module Spree
+  module SpreeMailchimpEcommerce
+    module ImageDecorator
+      def self.prepended(base)
+        base.after_create :update_mailchimp_product
+        base.after_update :update_mailchimp_product
+        base.after_destroy :update_mailchimp_product
+      end
+
+      private
+
+      def update_mailchimp_product
+        ::SpreeMailchimpEcommerce::UpdateProductJob.perform_later(find_product)
+      end
+
+      def find_product
+        Spree::Variant.find(self.viewable_id).product.id
```

Comment:
why not just `self.viewable`?

Reply:
For some reason it returns nil when calling viewable on Spree Image, thats why I added this.